staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -36,19 +36,40 @@ const (
 	familyIPv6 AddressFamily = 6
 )
 
+const (
+	IPv4RouteFile = "/proc/net/route"
+	IPv6RouteFile = "/proc/net/ipv6_route"
+)
+
 type Route struct {
 	Interface   string
 	Destination net.IP
 	Gateway     net.IP
-	// TODO: add more fields here if needed
+	Family      AddressFamily
 }
 
-// getRoutes obtains the IPv4 routes, and filters out non-default routes.
-func getRoutes(input io.Reader) ([]Route, error) {
-	routes := []Route{}
-	if input == nil {
-		return nil, fmt.Errorf("input is nil")
+type RouteFile struct {
+	name  string
+	parse func(input io.Reader) ([]Route, error)
+}
+
+var (
+	v4File = RouteFile{name: IPv4RouteFile, parse: getIPv4DefaultRoutes}
```

Comment:
Do you mean IPv4RouteFile/IPv6RouteFile? No, I'll make them private. 

If you are asking about v4File/v6File being at file scope, yes I need to be able to override them for unit tests. I can use strings to test the contents of route info files, but to test the upper level getAllDefaultRoutes(),with both positive and negative test cases, I need actual files, so I use file scope variables that I override for the tests. Could not figure out any other way to do that.